public/app/features/dashboard-scene/settings/annotations/AnnotationSettingsEdit.test.tsx

```diff
@@ -0,0 +1,278 @@
+import { of } from 'rxjs';
+
+import {
+  AnnotationQuery,
+  FieldType,
+  LoadingState,
+  PanelData,
+  VariableSupportType,
+  getDefaultTimeRange,
+  toDataFrame,
+} from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { setRunRequest } from '@grafana/runtime';
+import { mockDataSource } from 'app/features/alerting/unified/mocks';
+import { LegacyVariableQueryEditor } from 'app/features/variables/editor/LegacyVariableQueryEditor';
+
+import { AnnotationSettingsEdit } from './AnnotationSettingsEdit';
+
+const defaultDatasource = mockDataSource({
+  name: 'Default Test Data Source',
+  type: 'test',
+});
+
+const promDatasource = mockDataSource({
+  name: 'Prometheus',
+  type: 'prometheus',
+});
+
+jest.mock('@grafana/runtime', () => ({
+  ...jest.requireActual('@grafana/runtime'),
+  getAngularLoader: () => ({
+    load: () => ({
+      destroy: jest.fn(),
+      digest: jest.fn(),
+      getScope: () => ({
+        $watch: jest.fn(),
+      }),
+    }),
+  }),
+}));
+
+jest.mock('@grafana/runtime/src/services/dataSourceSrv', () => ({
+  ...jest.requireActual('@grafana/runtime/src/services/dataSourceSrv'),
+  getDataSourceSrv: () => ({
+    get: async () => ({
+      ...defaultDatasource,
+      variables: {
+        getType: () => VariableSupportType.Custom,
+        query: jest.fn(),
+        editor: jest.fn().mockImplementation(LegacyVariableQueryEditor),
+      },
+    }),
+    getList: () => [defaultDatasource, promDatasource],
+    getInstanceSettings: () => ({ ...defaultDatasource }),
+  }),
+}));
+
+jest.mock('./AngularEditorLoader', () => ({ AngularEditorLoader: () => 'mocked AngularEditorLoader' }));
+
+const runRequestMock = jest.fn().mockReturnValue(
+  of<PanelData>({
+    state: LoadingState.Done,
+    series: [
+      toDataFrame({
+        fields: [{ name: 'text', type: FieldType.string, values: ['val1', 'val2', 'val11'] }],
+      }),
+    ],
+    timeRange: getDefaultTimeRange(),
+  })
+);
+
+setRunRequest(runRequestMock);
+
+describe('AnnotationSettingsEdit', () => {
+  const mockOnUpdate = jest.fn();
+  const mockGoBackToList = jest.fn();
+  const mockOnDelete = jest.fn();
+  const mockOnPreview = jest.fn();
+
+  function setup() {
+    const annotationQuery: AnnotationQuery = {
+      name: 'test',
+      datasource: defaultDatasource,
+      enable: true,
+      hide: false,
+      iconColor: 'blue',
+    };
+
+    const props = {
+      annotation: annotationQuery,
+      onUpdate: mockOnUpdate,
+      editIndex: 1,
+      panels: [],
+      goBackToList: mockGoBackToList,
+      onDelete: mockOnDelete,
+      onPreview: mockOnPreview,
+    };
+
+    return {
+      anno: annotationQuery,
+      renderer: render(<AnnotationSettingsEdit {...props} />),
+      user: userEvent.setup(),
+    };
+  }
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render', async () => {
+    const {
+      renderer: { getByTestId },
+    } = setup();
+
+    await waitFor(() => {
```

Comment:
If you want to avoid all of `waitFor` you can always wrap the `render()` with `act()`, so you ensure the react component went through all the rendering cycles and it already ran all the async functions 😄 See [here](https://github.com/grafana/grafana/pull/81318/files#diff-cd58a512daf25cfb1e36bc87b1e667e0421f69aa88487a4255c5110ac8058935R72).

Just a tip, you don't need to change everything 🙏

Reply:
I'll look into this! Thank you!

Reply:
Refactored and remove `waitFor` calls! 💯